webapp/src/app.tsx

```diff
@@ -2059,12 +2059,19 @@ export class ProjectView
                         });
                     }
                 }
+                pxt.tickEvent("deploy.start")
                 return pxt.commands.deployAsync(resp, {
                     reportDeviceNotFoundAsync: (docPath, compileResult) => this.showDeviceNotFoundDialogAsync(docPath, compileResult),
                     reportError: (e) => core.errorNotification(e),
                     showNotification: (msg) => core.infoNotification(msg)
                 })
+                    .then(() => {
+                        pxt.tickEvent("deploy.finished")
+                    }, () => {
+                        pxt.tickEvent("deploy.unfinished")
+                    })
                     .catch(e => {
+                        pxt.tickEvent("deploy.exception")
```

Comment:
store whether "notifyUser" is set or not

Reply:
maybe use pxt.reportError so that we get a stacktrace as well.

Reply:
We use pxt.reportException below. I added the tick so it's easier to see the matching pairs of (deploy.started, deploy.finished) and (deploy.started, deploy.exception). Maybe it's my unfamiliarity with Kusto but I'm not sure how I'd correlate exceptions into (deploy.started, <some exception>) because the exceptions can be anything and don't have a prefix.

I moved the tickEvent to be right above the reportException so it's obvious what's happening.